### PR TITLE
Gradle 7 and Opensearch 2.0 upgrade

### PR DESCRIPTION
**Pull request by aksingh-es**
Signed-off-by: Amit Singh <snghes@amazon.com>

This PR support major campagins for Opensearch 2.0 release :
1.Using OS 2.0 snapshot
2.Remove mapping types ,transport deprecation support
3.Gradle 7 upgrade

Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/326
https://github.com/opensearch-project/opensearch-plugins/issues/130

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -549,40 +550,44 @@ check.dependsOn jacocoTestCoverageVerification
...
     // used for serializing/deserializing rcf models.
-    compile group: 'io.protostuff', name: 'protostuff-core', version: '1.8.0'
-    compile group: 'io.protostuff', name: 'protostuff-runtime', version: '1.8.0'
-    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
-
-    compile "org.jacoco:org.jacoco.agent:0.8.5"
-    compile ("org.jacoco:org.jacoco.ant:0.8.5") {
+    implementation group: 'io.protostuff', name: 'protostuff-core', version: '1.8.0'
+    implementation group: 'io.protostuff', name: 'protostuff-runtime', version: '1.8.0'
+    implementation group: 'io.protostuff', name: 'protostuff-api', version: '1.8.0'
```
Comment (ylwu-amzn) [inline build.gradle:579]: Why add these new dependencies: `protostuff-api` and `protostuff-collectionschema`?

Comment (aksingh-es) [reply build.gradle:579]: These required to be explicitly added as the compilation errors for Linkedbuffer and Schema  were coming from these.
Compile failures : https://paste.amazon.com/show/snghes/1647551203

File: build.gradle [hunk 2]
```diff
@@ -591,11 +596,11 @@ dependencies {
...
     testImplementation group: 'org.powermock', name: 'powermock-reflect', version: '2.0.7'
     testImplementation group: 'org.objenesis', name: 'objenesis', version: '3.0.1'
-    testCompile group: 'net.bytebuddy', name: 'byte-buddy', version: '1.9.15'
-    testCompile group: 'net.bytebuddy', name: 'byte-buddy-agent', version: '1.9.15'
+    testImplementation group: 'net.bytebuddy', name: 'byte-buddy', version: '1.9.15'
+    testImplementation group: 'net.bytebuddy', name: 'byte-buddy-agent', version: '1.9.15'
     testCompileOnly 'org.apiguardian:apiguardian-api:1.1.0'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.2'
-    testImplementation 'org.junit.jupiter:junit-jupiter-params:5.7.2'
+    testImplementation  'org.junit.jupiter:junit-jupiter-params:5.7.2'
```
Comment (ylwu-amzn) [inline build.gradle:603]: minor: extra space

**Review by ylwu-amzn — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -29,63 +29,64 @@ jobs:
...
+#           security=`curl -XGET https://localhost:9200/_cat/plugins?v -u admin:admin --insecure |grep opensearch-security|wc -l`
+#           if [ $security -gt 0 ]
+#           then
+#             echo "Security plugin is available"
+#             ./gradlew integTest -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster" -Dhttps=true -Duser=admin -Dpassword=admin
+#           else
+#             echo "Security plugin is NOT available"
+#             ./gradlew integTest -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster"
+#           fi
+#       - name: Run AD Backwards Compatibility Tests
```
Comment (amitgalitz) [inline .github/workflows/CI.yml:86]: Should this also be commented for now?

File: gradlew [hunk 4]
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (amitgalitz) [inline gradlew:47]: Is there any context for these new memory defaults?

File: src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java [hunk 5]
```diff
@@ -919,10 +919,7 @@ private void updateMappingIfNecessary(GroupedActionListener<Void> delegateListen
                     adminClient
                         .indices()
                         .putMapping(
-                            new PutMappingRequest()
-                                .indices(adIndex.getIndexName())
-                                .type(CommonName.MAPPING_TYPE)
```
Comment (amitgalitz) [inline src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java:924]: if all uses of `CommonName.Mapping_TYPE` were removed we can maybe delete that constant and comment referring elastic above it in `CommonName` class

File: .github/workflows/CI.yml (hunk 3 above)
Comment (aksingh-es) [reply .github/workflows/CI.yml:86]: There is depenedency on job-scheduler-2.0.0.0..which is currently alpha stage.Will be better to uncomment together hen we have the tested job-scheduler code

File: gradlew (hunk 4 above)
Comment (aksingh-es) [reply gradlew:47]: these are auto generated

File: src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java (hunk 5 above)
Comment (aksingh-es) [reply src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java:924]: Good call out.would do that in separate PR.This is already many files

Comment (amitgalitz) [reply src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java:924]: Sounds good

**Review by amitgalitz — APPROVED** [review]
LGTM! Thanks for all the changes, lets keep an eye out for any updates on job-scheduler and other 2.0 dependency to enable some parts of CI again

**Pull request merged**